### PR TITLE
[apinotes] Rename NSApplication activateIgnoringOtherApps(_:) to activate(ignoringOtherApps:)

### PR DESCRIPTION
**Pull request by jtbandes**
#### What's in this pull request?

Renaming `activateIgnoringOtherApps(_:)` to `activate(ignoringOtherApps:)`.

I'm not sure if changes like this are supposed to go through swift-evolution? It seems like a pretty obvious one to me. cc @moiseev who has worked on the overlay before

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by moiseev** [comment]
Let me discuss this change with the AppKit people.

File: stdlib/public/SDK/AppKit/AppKit.swift [hunk 1]
```diff
@@ -141,3 +141,10 @@ extension NSWindowController {
     self.dismissController(sender)
   }
 }
+
+extension NSApplication {
```
Comment (jrose-apple) [inline stdlib/public/SDK/AppKit/AppKit.swift]: This part wouldn't be necessary; the compiler generates stubs for the name as imported in Swift 2.

Comment (jtbandes) [inline stdlib/public/SDK/AppKit/AppKit.swift]: Oh, ok. Then I wonder why some of the other methods here exist, when they could have just been done with APINotes?

**Comment by moiseev** [comment]
After having discussed the proposed change with the API owners, we will accept the PR.

One thing to note though, since changes like this one affect the framework APIs, a proper discussion should be triggered by a radar. So please, file a radar if you (and it's not only you, @jtbandes) think that some APIs can be improved for Swift.

Thanks, Jacob for bringing this one up!

**Comment by moiseev** [comment]
@swift-ci Please test and merge

**Comment by jtbandes** [comment]
Thank you!  For completeness, I've just filed rdar://27556394.

**Comment by jtbandes** [comment]
Failures unrelated.

**Comment by jtbandes** [comment]
@moiseev This is worth another shot now that CI builds are working.

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge.

**Comment by moiseev** [comment]
Thanks, @CodaFi

**Comment by CodaFi** [comment]
I think so too.  Let's try again

@swift-ci please smoke test and merge.

**Comment by shahmishal** [comment]
@CodaFi Sorry, at this time "Please smoke test and merge" is not supported. Please use @swift-ci Please smoke test

**Comment by tkremenek** [comment]
@swift-ci smoke test

**Comment by tkremenek** [comment]
@swift-ci smoke test linux

**Comment by gottesmm** [comment]
@tkremenek Did you mean to start the smoke test? (Just asking b/c you just merged). If not I would like to kill the job.

**Comment by tkremenek** [comment]
@gottesmm I started the smoke test for Linux, and realized it was not needed because this does not impact Linux.

**Comment by gottesmm** [comment]
@tkremenek Ok. I will kill the job.

**Comment by tkremenek** [comment]
@gottesmm thanks.

**Comment by jtbandes** [comment]
Thanks all 😊

**Pull request merged**